docs/src/token-2022/extensions.md

```diff
@@ -2,4 +2,997 @@
+
+<Tabs className="unique-tabs" groupId="language-selection">
+  <TabItem value="cli" label="CLI" default>
+
+CLI support coming soon!
+
+  </TabItem>
+  <TabItem value="jsx" label="JS">
+
+```jsx
+import {
+    clusterApiUrl,
+    sendAndConfirmTransaction,
+    Connection,
+    Keypair,
+    SystemProgram,
+    Transaction,
+    LAMPORTS_PER_SOL,
+} from '@solana/web3.js';
+import {
+    createAccount,
+    createMint,
+    createInitializeImmutableOwnerInstruction,
+    createInitializeAccountInstruction,
+    getAccountLen,
+    ExtensionType,
+    TOKEN_2022_PROGRAM_ID,
+} from '@solana/spl-token';
+
+(async () => {
+    const connection = new Connection(clusterApiUrl('devnet'), 'confirmed');
+
+    const payer = Keypair.generate();
+    const airdropSignature = await connection.requestAirdrop(payer.publicKey, 2 * LAMPORTS_PER_SOL);
+    await connection.confirmTransaction(airdropSignature);
+
+    const mintAuthority = Keypair.generate();
+    const decimals = 9;
+    const mint = await createMint(
+        connection,
+        payer,
+        mintAuthority.publicKey,
+        mintAuthority.publicKey,
+        decimals,
+        undefined,
+        undefined,
+        TOKEN_2022_PROGRAM_ID
+    );
+
+    const accountLen = getAccountLen([ExtensionType.ImmutableOwner]);
+    const lamports = await connection.getMinimumBalanceForRentExemption(accountLen);
+
+    const owner = Keypair.generate();
+    const accountKeypair = Keypair.generate();
+    const account = accountKeypair.publicKey;
+    const transaction = new Transaction().add(
+        SystemProgram.createAccount({
+            fromPubkey: payer.publicKey,
+            newAccountPubkey: account,
+            space: accountLen,
+            lamports,
+            programId: TOKEN_2022_PROGRAM_ID,
+        }),
+        createInitializeImmutableOwnerInstruction(account, TOKEN_2022_PROGRAM_ID),
+        createInitializeAccountInstruction(account, mint, owner.publicKey, TOKEN_2022_PROGRAM_ID)
+    );
+    await sendAndConfirmTransaction(connection, transaction, [payer, accountKeypair], undefined);
+})();
+```
+
+  </TabItem>
+</Tabs>
+
+#### Example: Creating an associated token account with immutable ownership
+
+All associated token accounts have the immutable owner extension included, so
+it's extremely easy to use the extension.
+
+<Tabs className="unique-tabs" groupId="language-selection">
+  <TabItem value="cli" label="CLI" default>
+
+CLI support coming soon!
+
+  </TabItem>
+  <TabItem value="jsx" label="JS">
+
+```jsx
+    const associatedAccount = await createAccount(
+        connection,
+        payer,
+        mint,
+        owner.publicKey,
+        undefined,
+        undefined,
+        TOKEN_2022_PROGRAM_ID
+    );
+```
+
+  </TabItem>
+</Tabs>
+
+### Non-Transferable Tokens
+
+To accompany immutably owner token accounts, the `NonTransferable` mint extension
+allows for "soul-bound" tokens that cannot be moved to any other entity. For
```

Comment:
I know "soul-bound" is the recent common term used in the Eth ecosystem, but do we want to use it here too? Maybe a #terminology thing

Reply:
Up to you! I always think of "soul-bound" from my MMO days :laughing:

Reply:
We can stick with it until we have a better recommendation imo